### PR TITLE
Extra spaces between observations

### PR DESCRIPTION
**Pull request by cquiroz**
Before:
image

After:
image

### CODE REVIEW

**Review by rpiaggio — APPROVED** [review]
LGTM, but out of curiosity: why is the ODB returning deleted observations if the default is `includeDeleted: Boolean! = false` ?

**Comment by cquiroz** [comment]
I also wondered that as well. We have two queries one for obs which works as expected but the one getting groups brings them all, I couldn't find a way to filter them

File: explore/src/main/scala/explore/observationtree/ObsList.scala [hunk 1]
```diff
@@ -352,7 +352,14 @@ object ObsList:
               <.div(
                 ^.overflow := "auto",
                 Tree(
-                  treeNodes.get,
+                  treeNodes.get.filter(_.data match {
+                    case Left(GroupTree.Obs(id)) =>
+                      props.observations.get
+                        .getValue(id)
+                        .map(o => !o.deleted)
+                        .getOrElse(false)
```
Comment (hugo-vrijswijk) [inline explore/src/main/scala/explore/observationtree/ObsList.scala:360]:
```suggestion
                        .exists(o => !o.deleted)
```

**Review by hugo-vrijswijk — COMMENTED** [review]
Interesting, I thought nodes without existence were already filtered out in GroupTree.scala. Maybe it makes more sense to fix the bug there and not create the nodes in the first place?

Update:
We do check for existence, but I forgot to add the check for root groups. 

image

**Comment by cquiroz** [comment]
> Interesting, I thought nodes without existence were already filtered out in GroupTree.scala. Maybe it makes more sense to fix the bug there and not create the nodes in the first place?
> 
> Update: We do check for existence, but I forgot to add the check for root groups.
> image

Do you think it maybe better to fix it this way?
We can also consider whether the backend should not return deleted nodes in the first place. It would imply less traffic as well

**Comment by hugo-vrijswijk** [comment]
> > Interesting, I thought nodes without existence were already filtered out in GroupTree.scala. Maybe it makes more sense to fix the bug there and not create the nodes in the first place?
> > Update: We do check for existence, but I forgot to add the check for root groups.
> > image
> 
> Do you think it maybe better to fix it this way? We can also consider whether the backend should not return deleted nodes in the first place. It would imply less traffic as well

It seems like a cleaner solution to not _create_ the nodes we don't want to show, rather than trying to hide them and adding extra CSS. Existence is already queried in the group elements query, so that saves adding it to the query in this PR. 

I think the cleanest would definitely be not returning non-existent elements in the group elements query.

**Comment by cquiroz** [comment]
I agree, let's make a sample query to let the odb team fix it

**Comment by cquiroz** [comment]
https://github.com/gemini-hlsw/lucuma-odb/issues/1182

**Comment by cquiroz** [comment]
This was fixing upstream, closing...

**Pull request closed without merging**